ai_document_search_backend/endpoints.py

```diff
@@ -12,15 +13,76 @@ class SummarizationResponse(BaseModel):
     summary: str
 
 
+class User(BaseModel):
```

Comment:
Would be nice to put all auth logic to an `/auth` router in a separate file, as per https://fastapi.tiangolo.com/tutorial/bigger-applications/.